gif-screencast.el

```diff
@@ -121,13 +121,83 @@ various programs run here."
   :init-value nil
   :keymap gif-screencast-mode-map)
 
+(defvar gif-screencast--done nil
+  "Flag if capturing already done.")
+
+(defvar gif-screencast--finishing nil
+  "Flag if compilation in progress.")
+
+(defvar gif-screencast--counter 0
+  "Number of running screenshots.")
+
+(defun gif-screencast-capture-sentinel (_proc _status)
+  "Sentinel for screen capturing."
+  (setq gif-screencast--counter (- gif-screencast--counter 1))
+  (gif-screencast--finish))
+
+(defun gif-screencast-done ()
+  "Return t if screen capturing already done."
+  (and gif-screencast--done (= gif-screencast--counter 0)))
+
+(defun gif-screencast--finish ()
+  "Finish screen capturing."
+  (if (gif-screencast-done)
+      (if gif-screencast--finishing
+          nil
+        (setq gif-screencast--finishing t)
+        (let (delays
+              (index 0)
+              (frames gif-screencast--frames))
+          (while (cdr frames)
+            (push (list "(" "-clone" (number-to-string index) "-set" "delay"
+                        ;; Converters delays are expressed in centiseconds.
+                        (format "%d" (* 100 (float-time
+                                             (time-subtract (car (cadr frames)) (caar frames)))))
+                        ")" "-swap" (number-to-string index) "+delete")
+                  delays)
+            (setq index (1+ index)
+                  frames (cdr frames)))
+          (message "Compiling GIF with %s..." gif-screencast-convert-program)
+          (let ((output (expand-file-name
+                         (format-time-string "output-%F-%T.gif" (current-time))
+                         (or (and  (file-writable-p gif-screencast-output-directory)
+                                   gif-screencast-output-directory)
+                             (read-directory-name "Save output to directory: ")))))
+            (setq p (apply 'start-process
```

Comment:
This `p` should be let-bound.

Reply:
It's copy paste of your code - I don't change it. It need more testing for changes like this.

Reply:
It was a mistake on my end.  I'll change it.